src/java/htsjdk/variant/variantcontext/filter/HetrozygosityFilter.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2015 The Broad Institute
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package htsjdk.variant.variantcontext.filter;
+
+import htsjdk.variant.variantcontext.Genotype;
+import htsjdk.variant.variantcontext.VariantContext;
+
+/**
+ * A Predicate on VariantContexts that either returns true at hetrozygous sites (invertible to false).
+ * if optional "sample" argument to constructor is given, the genotype of that sample will be examined,
+ * otherwise first genotype will be used.
+ *
+ * Missing sample, or no genotype will result in an exception being thrown.
+ */
+public class HetrozygosityFilter implements VariantContextFilter {
+
+    private String sample;
+    private boolean keepHets;
+
+    /**
+     * Constructor
+     *
+     * @param keepHets determine whether to keep the het sites ( true ) or filter them out ( false )
+     * @param sample the name of the sample in the variant context whose genotype should be examined.
+     */
+    public HetrozygosityFilter(boolean keepHets, String sample) {
+        this.keepHets = keepHets;
+        this.sample = sample;
+    }
+
+    public HetrozygosityFilter(boolean keepHets) {
+        this(keepHets, null);
+    }
+
+    /* @return true if record is not be kept, otherwise false
+    * Assumes that <sample> is a sample in the vcf, if not null,
+    * otherwise looks for the first genotype (and assumes it exists).
+    * */
+    @Override
+    public boolean filterOut(final VariantContext record) {
```

Comment:
that's why I changed getGenotype(int) so that it will return null if there's no such genotype.

test for this case is provided for here:
https://github.com/samtools/htsjdk/pull/306/files#diff-6a4feb02d63cca89ea7d6ee0f00897e6R81